### DISCUSSION TITLE
Reset method

### DISCUSSION THREAD

**Discussion by HaNdTriX** [TanStack/query#431] [General]
Apollo has two methods named: `apolloClient.clearStore` && `apolloClient.resetStore`.

I want to achieve something similar. Take the following example.
On logout I want to clear the cache and refetch all queries immediately without my ui getting stale.

```js
import { useCallback } from "react";
import { queryCache } from "react-query";

export default function useLogout() {
  return useCallback(() => {
    // 1. Remove cookie
    document.cookie = `token=; Max-Age=-99999999;`;

    // 2. Clear cache (refetch all data)
    queryCache.clear()
  }, []);
}
```

Currently `queryCache.clear()` refetches all queries but doesn't remove the stale data from the UI. Instead it waits for the refetched queries to resolve. 

Is there a way to reset all queries back to a loading state without having stale data?

**Comment by tannerlinsley** [comment]
I'll look into this. We may need to add a new method

**Reply by ahmetuysal** [reply]
Any updates on this? I looked into docs but couldn't find something similar. Thank you for your amazing work!

**Reply by znareak** [reply]
I actually need something like this, my react U has old data and react-query keeps the cache in its memory.
removeQueries or invalidateQueries dont work for me.
Thank you for you support.

**Comment by italodeandra** [comment]
@tannerlinsley Is there any workaround for this issue?

**Comment by aaronjensen** [comment]
This is also being discussed here and a PR is in the works:

https://github.com/tannerlinsley/react-query/discussions/1363#discussioncomment-166785
https://github.com/tannerlinsley/react-query/pull/1375
